modules/core/src/test/java/org/apache/ignite/internal/processors/jobmetrics/GridJobMetricsSelfTest.java

```diff
@@ -0,0 +1,358 @@
+        }
+    }
+
+    /** Test correct calculation of finished, started, active, canceled metrics of the {@link GridJobProcessor}. */
+    @Test
+    public void testGridJobMetrics() throws Exception {
+        latch = new CountDownLatch(1);
+
+        try(IgniteEx g = startGrid(0)) {
+            ReadOnlyMetricRegistry mreg = g.context().metric().registry().withPrefix(JOBS);
+
+            LongMetric started = (LongMetric)mreg.findMetric(STARTED);
+            LongMetric active = (LongMetric)mreg.findMetric(ACTIVE);
+            LongMetric waiting = (LongMetric)mreg.findMetric(WAITING);
+            LongMetric canceled = (LongMetric)mreg.findMetric(CANCELED);
+            LongMetric rejected = (LongMetric)mreg.findMetric(REJECTED);
+            LongMetric finished = (LongMetric)mreg.findMetric(FINISHED);
+            LongMetric totalExecutionTime = (LongMetric)mreg.findMetric(EXECUTION_TIME);
+            LongMetric totalWaitingTime = (LongMetric)mreg.findMetric(WAITING_TIME);
+
+            assertNotNull(started);
+            assertNotNull(active);
+            assertNotNull(waiting);
+            assertNotNull(canceled);
+            assertNotNull(rejected);
+            assertNotNull(finished);
+            assertNotNull(totalExecutionTime);
+            assertNotNull(totalWaitingTime);
+
+            assertEquals(0, started.value());
+            assertEquals(0, active.value());
+            assertEquals(0, waiting.value());
+            assertEquals(0, canceled.value());
+            assertEquals(0, rejected.value());
+            assertEquals(0, finished.value());
+            assertEquals(0, totalExecutionTime.value());
+            assertEquals(0, totalWaitingTime.value());
+
+            SimplestTask task = new SimplestTask();
+
+            g.compute().execute(task, 1);
+
+            //Waiting task to finish.
+            boolean res = waitForCondition(() -> active.value() == 0, TIMEOUT);
+
+            assertTrue("Active = " + active.value(), res);
+
+            assertEquals(1, started.value());
+            assertEquals(0, waiting.value());
+            assertEquals(0, canceled.value());
+            assertEquals(0, rejected.value());
+            assertEquals(1, finished.value());
+
+            //Task should block until latch is down.
+            task.block = true;
+
+            ComputeTaskFuture<?> fut = g.compute().executeAsync(task, 1);
+
+            //Waiting task to start execution.
+            res = waitForCondition(() -> active.value() == 1, TIMEOUT);
+
+            assertTrue("Active = " + active.value(), res);
+
+            assertEquals(2, started.value());
+            assertEquals(0, waiting.value());
+            assertEquals(0, canceled.value());
+            assertEquals(0, rejected.value());
+            assertEquals(1, finished.value());
+
+            res = waitForCondition(() -> active.value() > 0, TIMEOUT);
```

Comment:
There nothing happened between wait for `active.value() == 1` and wait for `active.value() > 0`. Why should we wait twice?

Reply:
1. `ComputeTaskFuture<?> fut = g.compute().executeAsync(task, 1);` - starting task invocation that will be waiting on `latch` for imitaion of long running job.
2. `res = waitForCondition(() -> active.value() == 1, TIMEOUT);` - waiting this job to start.
3. `latch.countDown();` - now started job can unblock and finish.
4. `res = waitForCondition(() -> active.value() == 0, TIMEOUT);` - waiting for metric update.

Reply:
OK, now I see - the question is about the second assert of `active.value() > 0` :)
Seems, it's duplicate, as you wrote.
Removed.
Thanks!